feat(lanelet2_extension)!: introduce API versioning along with format_version

soblin opened the pull request:
## Description

https://github.com/orgs/autowarefoundation/discussions/4740

## Related links

https://github.com/orgs/autowarefoundation/discussions/4740

## Tests performed

https://evaluation.tier4.jp/evaluation/reports/50c7b099-3375-5eb9-8fd5-710fe2011a9c?project_id=prd_jt

## Notes for reviewers

<!-- Write additional information if necessary. It should be written if there are related PRs that should be merged at the same time. -->

## Interface changes

introduce `inline namespace v1`

## Effects on system behavior

none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

youtalk commented on autoware_lanelet2_extension/lib/query.cpp:
```diff
@@ -50,64 +50,27 @@
 using lanelet::utils::to2D;
 
-namespace lanelet::utils
+namespace lanelet::utils::query
```
This change seems to be a refactoring that is unrelated to the `format_version`. If it's true, I think it should not be included in this PR to clarify the modifications made in this PR.

youtalk commented on autoware_lanelet2_extension/lib/visualization.cpp:
```diff
@@ -346,158 +346,12 @@ void pushPolygonMarker(
 }  // anonymous namespace
 
-namespace lanelet
+namespace lanelet::visualization
```
It's the same as above.

youtalk reviewed (commented):
The name `namespace v1` did not align with the version in `package.xml` and felt unclear. Instead of forcing a short name, how about using descriptive name like `lanelet_format_v1` or `format_v1`?

soblin commented:
@youtalk That's right, I'd prefer `format_v1` because it is shorter and we can discriminate it from package version itself !

soblin commented:
@youtalk After your suggestions I:
- replaced `v1` with `format_v1`
- moved `inline namespace format_v1` to top. So some functions have moved
- only modify the existing functions for the above changes

youtalk commented on autoware_lanelet2_extension/include/autoware_lanelet2_extension/version.hpp:
```diff
@@ -0,0 +1,36 @@
+ * @brief denotes the major format_version
+ */
+enum class Version : int {
+  none = 0,
```
```suggestion
  format_none = 0,
```
is better?

youtalk reviewed (approved):
Thank you for your update. It was very easy to review again.
